src/library/Pagination/Pagination.js

```diff
@@ -0,0 +1,248 @@
+/* @flow */
+import React, { PureComponent } from 'react';
+import { createStyledComponent } from '../styles';
+import { withTheme } from '../themes';
+import Flex from '../Flex';
+import Pages from './Pages';
+import PageJumper from './PageJumper';
+import _PageSizer from './PageSizer';
+
+type Props = {
+  /**
+   * The category representing items to be paginated; utilized in
+   * [page sizer](#page-sizer) caption
+   */
+  category?: string,
+  /** The currently displayed page */
+  currentPage: number,
+  /** Hide the page number buttons */
+  showPageNumbers?: boolean,
+  /**
+   * Various messages and labels used by Pagination
+   * ([see example for more details](#rtl))
+   */
+  messages: Messages,
+  /** Called when current page is changed */
+  onPageChange: (currentPage: number) => void,
+  /** Called when [page size](#page-sizer) is changed */
+  onPageSizeChange?: (pageSize: number) => void,
+  /**
+   * Render a [page jumper](#page-jumper)
+   * ([TextInput component](/components/text-input)); enables the user to change
+   * the current page on number entry
+   */
+  showPageJumper?: boolean,
+  /** The number of items or data to be rendered on each page */
+  pageSize: number,
+  /**
+   * Render a [page sizer](#page-sizer)
+   * ([Select component](/components/select)); enables the user to select the
+   * page size
+   */
+  showPageSizer?: boolean,
+  /**
+   * A collection of possible page sizes for a user to select from; implemented
+   * in [page sizer](#page-sizer). Be sure to provide unique values.
+   */
+  pageSizes?: Array<number>,
+  /** Available sizes */
+  size?: 'small' | 'medium' | 'large' | 'jumbo',
+  /** @Private */
+  theme: Object,
+  /** The total number of items or data to be paginated */
+  totalCount: number,
+  /**
+   * The number of consecutive pages to be displayed when the paginated results
+   * are truncated ([see example for details](#visible-range))
+   */
+  visibleRange?: number
+};
+
+export type Messages = {
+  pagination: { label: string, next: string, previous: string },
+  showPageJumper?: { label: string, placeholder: string },
+  showPageSizer?: {
+    status: (
+      category: string,
+      first: number,
+      last: number,
+      total: number
+    ) => string,
+    itemText: (pageSize: number) => string
+  }
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  PaginationPageJumper_width: '4.65em',
+  Pagination_gutterWidth: baseTheme.space_inline_sm,
+  ...baseTheme
+});
+
+const styles = {
+  pageSizer: ({ theme }) => {
```

Comment:
If there is only one item in styles, I'd prolly just inline these in the definition of PageSizer.

Reply:
Styles should've also had `root` styles; it's updated now.